design/spec/upsert/protocol.md

```
@@ -0,0 +1,166 @@
+> Internal lock protocol for `upsert` (#847): try_upsert, try_upsert_erase, version validation, retry logic.
```

markdownlint-cli2 (0.22.1)

[warning] First line in a file should be a top-level heading

(MD041, first-line-heading, first-line-h1)